core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -1131,14 +1131,47 @@ private[spark] class BlockManager(
   }
 
   /**
+   * Called for pro-active replenishment of blocks lost due to executor failures
+   *
+   * @param blockId blockId being replicate
+   * @param replicas existing block managers that have a replica
+   * @param maxReps maximum replicas needed
+   * @return
+   */
+  def replicateBlock(blockId: BlockId, replicas: Set[BlockManagerId], maxReps: Int): Boolean = {
+    logInfo(s"Pro-actively replicating $blockId")
+    val infoForReplication = blockInfoManager.lockForReading(blockId).map { info =>
+      val data = doGetLocalBytes(blockId, info)
+      val storageLevel = StorageLevel(
+        info.level.useDisk,
+        info.level.useMemory,
+        info.level.useOffHeap,
+        info.level.deserialized,
+        maxReps)
+      (data, storageLevel, info.classTag)
+    }
+    infoForReplication.foreach { case (data, storageLevel, classTag) =>
+      replicate(blockId, data, storageLevel, classTag, replicas)
+    }
+    true
+  }
+
+  /**
    * Replicate block to another node. Note that this is a blocking call that returns after
    * the block has been replicated.
+   *
+   * @param blockId
+   * @param data
+   * @param level
+   * @param classTag
+   * @param existingReplicas
    */
   private def replicate(
-      blockId: BlockId,
-      data: ChunkedByteBuffer,
-      level: StorageLevel,
-      classTag: ClassTag[_]): Unit = {
```

Comment:
@sameeragarwal This code is being removed as a part of this PR. Code replacing this has this fixed.